### ISSUE TITLE
Complex Encoder-Decoder System for Titanic like Data

### ISSUE THREAD

**Issue by GaganaB** [uber/ludwig#690]
Hi, 

I was wondering if there is a way to use complex encoder system for titanic-like binary classification? What would you recommend to enhance performance on this data? 

Thank you.

**Comment by w4nderlust** [comment]
Hi @GaganaB , I'm not really sure what do you mean by "complex encoder system". My best guess is that it means a combination of parameters for the encoders of each feature that can squeeze more performance on that dataset, am I correct?
I don't have an easy answer here, because i haven't spent much time optimizing for that dataset specifically (as it is kinda pointless anyway, given the toy nature of the problem), but there are a few things I can point out that could be useful:
- change some of the preprocessing parameters for numerical features, for instance use z-score or minmax normalization
- play around with the combiner parameters, specifically the number and size of layers. If you make them too big, the model will likely overfit, but you can compensate by increasing regularization and dropout
- play around with the threshold of the binary output feature, as the best f1 my come from either increasing or decreasing it to either improve accuracy or recall

Moreover, we are considering adding additional preprocessing for numerical features, binning and binarization in particular, as it has been shown in some cases that by doing so NN models can outperform tree based models also on tabular data, see keep an eye out for when we are going to add it to Ludwig!

**Comment by GaganaB** [comment]
Thanks a lot. I think this works. Is there a way in Ludwig to try model ensembling?

**Comment by w4nderlust** [comment]
@GaganaB you are welocme. And no there's no ensembling mechanism within Ludwig, you can easily build your own as training of Ludwig models in the API can be done in 2 lines of code, you can train a bunch, collect their predictions and probabilities and then train another one on top of those. Not sure doing it for the Titanic dataset makes sense though :)
